_posts/2020-03-19-our-rails-upgrade-process.markdown

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: "Our Rails Upgrade Process"
+date: 2020-03-17 10:30:00
+reviewed: 2020-03-19 10:00:00
+categories: ["rails", "upgrade"]
+author: "cleiviane"
+---
+
+We know that there are many challenges involving a Rails upgrade project. Depending on how big is your application, how old is your Rails version and how well structured your code is, it can be difficult to perform that job and keep your sanity. If you don't find a reliable and trustable process to guide you from version X to version Y, you can end-up in a nightmare.
+
+The good news is that here at [Ombu Labs](https://www.ombulabs.com/), we have been upgrading Rails applications for over 10 years now and this gave us a know-how to define a process that has proven to be very effective. And today I want to share Our Rails Upgrade Process with you.
+
+<!--more-->
+
+## Step 1: Finding deprecation warnings
+
+The first thing that we do is to search for deprecation warnings in the application, which are log messages suggesting things that you will need to change when moving for the next version of Rails. That can be because something will either be removed from the source code or the code will work in a different way.
+
+You might find these deprecation warnings by searching at the `production.log` or tracking then down in your Log Management Software, such as [Sentry](https://sentry.io/) or [Splunk](https://www.splunk.com).
+
```

Comment:
You can also search for deprecation warnings in test.log, so you can find them in your CI service as well.